### PR TITLE
Update banner after bid

### PR DESCRIPTION
**Pull request by erikdstock**
Tracked in https://artsyproduct.atlassian.net/browse/PURCHASE-248

@ashfurrow helped me with the first two commits of this. It seems to be working now (see below- the banner should update after bidding and going from outbid to winning) but there is an error I've seen before [see below].
bannerupdate 

[resolved]
```
2018-07-17 14:36:32.614393+0200 Artsy[83465:8377853] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   Foundation                          0x000000010e626df7 _AssertAutolayoutOnAllowedThreadsOnly + 77
	1   Foundation                          0x000000010e432540 -[NSISEngine withBehaviors:performModifications:] + 28
	2   Foundation                          0x000000010e42dd67 +[NSLayoutConstraint _addOrRemoveConstraints:activate:] + 403
	3   UIKit                               0x00000001111b174b _UIViewRemoveConstraintsMadeDanglyByChangingSuperview + 1189
	4   UIKit                               0x00000001106f505a __45-[UIView(Hierarchy) _postMovedFromSuperview:]_block_invoke + 61
	5   UIKit                               0x00000001106f4f8a -[UIView(Hierarchy) _postMovedFromSuperview:] + 808
	6   UIKit                               0x00000001106f2e3e __UIViewWasRemovedFromSuperview + 169
	7   UIKit                               0x00000001106f2930 -[UIView(Hierarchy) removeFromSuperview] + 479
	8   Artsy                               0x000000010c4c5b14 -[ORStackView removeSubview:] + 212
	9   Artsy                               0x000000010bfdb547 -[ARArtworkActionsView artworkBidUpdated:] + 647
	10  CoreFoundation  
```

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/Views/Artwork/ARArtworkMetadataView.h [hunk 1]
```diff
@@ -22,8 +22,11 @@
 
 - (void)setDelegate:(id<ARArtworkDetailViewDelegate, ARArtworkDetailViewButtonDelegate, ARArtworkActionsViewDelegate, ARArtworkActionsViewButtonDelegate, ARArtworkPreviewImageViewDelegate, ARArtworkPreviewActionsViewDelegate>)delegate;
 
+- (void)updateUI;
+
 @property (nonatomic, strong, readonly) UIView *left;
 @property (nonatomic, strong, readonly) UIView *right;
+@property (nonatomic, strong) ARArtworkActionsView *actionsView;
```
Comment (ashfurrow) [inline Artsy/Views/Artwork/ARArtworkMetadataView.h]: It might make sense to keep this property only defined in the `.m` file and expose a `showSpinner` or `showAuctionSpinner` method to call instead.

File: Artsy/Views/Artwork/ARArtworkActionsView.m [hunk 2]
```diff
@@ -90,11 +90,8 @@ - (void)artworkBidUpdated:(NSNotification *)notification;
         for (UIView *subview in self.subviews) {
```
Comment (ashfurrow) [inline Artsy/Views/Artwork/ARArtworkActionsView.m]: My thinking here was that, since `ARArtworkView` is already listening for `ARAuctionArtworkBidUpdatedNotification`, do we need the `artworkBidUpdated:` method in `ARArtworkActionsView` at all? I think we could delete it at this point and things would work.

File: Artsy/Views/Artwork/ARArtworkView.m [hunk 3]
```diff
@@ -148,6 +151,33 @@ - (void)setUpCallbacks
...
+                sself.banner.auctionState = saleArtwork.auctionState;
+                [UIView animateIf:ARPerformWorkAsynchronously duration:ARAnimationDuration :^{
+                    [sself.banner updateHeightConstraint];
+                    [sself.stackView layoutIfNeeded];
+                }];
+                [sself.metadataView updateUI];
+            }
+        }
+      failure:nil
+        allowCached:NO];
```
Comment (alloy) [inline Artsy/Views/Artwork/ARArtworkView.m]: These lines are a bit hard to parse.

Comment (erikdstock) [reply Artsy/Views/Artwork/ARArtworkView.m]: Should this just be indented to line up with `onSaleArtworkUpdate` on L165 (that is, very indented)? I don't know the convention with objC and xcode kept correcting it to this.

Comment (alloy) [reply Artsy/Views/Artwork/ARArtworkView.m]: Yeah that’s what, in my experience, is ideal objc formatting. Sigh, Xcode. I wouldn’t battle too much with that, though, Xcode always wins.

Comment (erikdstock) [reply Artsy/Views/Artwork/ARArtworkView.m]: Once you made this comment and I looked at it that indentation became clear to me. I just wasn't thinking in these terms. Thanks for pointing it out.

**Comment by erikdstock** [comment]
merge on green
